MIGRATION_v4x.md

```diff
@@ -0,0 +1,158 @@
+# Migration Guide
+
+**Table of Contents**
+
+- [MUI 5 Migration](#mui-5-migration)
+- [Packages Changes](#packages-changes)
+  - [Visualizations package](#visualizations-package)
+  - [Compatibility package](#compatibility-package)
+- [Components](#components)
+- [What's next?](#whats-next)
+
+# From v3 to v4
+
+Having reached a point of stability and maturity we have approached v4 mostly as a **technical release**, with a focus on cleanup and organization in order to **unlock the next stage of growth** (streamline the adoption of new versions of the Design System and improve the customization experience).
+
+## MUI 5 Migration
+
+We have been following [MUI](https://mui.com/) for a long time and using many of the solutions provided since day one. Our approach has always been to use what we find value and unlock the remaining features for our users. Following the same spirit we decided that it was important to upgrade to the latest MUI v5.
+
+- **Styling Solution** - This is our biggest coupling point with MUI today since we are leveraging entirely their [JSS styling system](https://mui.com/system/styles/basics/). For many reasons MUI 5 moved from [JSS to emotion](https://mui.com/blog/mui-core-v5/#migration-from-jss-to-emotion), but since we could keep using JSS, we thought it would be a perfect compromise for now, since it gives us the necessary time to do our own evaluation.
+
+### Update MUI packages
+
+Install the Material UI v5 packages.
```

Comment:
Somewhere around this section we should highlight the changes to the typescript definition of the theme, each project may need to create an override of the material theme to include our .hv value otherwise the app may work but typescript is not going to recognize the theme